### PR TITLE
Issue154 upgrade smart contracts to provwasm 2.0.0

### PR DESCRIPTION
**Pull request by leeduan**
Migration guide: https://github.com/provenance-io/provwasm/blob/main/MIGRATION.md

There are a couple broken tests right now that are waiting on this discussion: https://discord.com/channels/897185242182484029/1115703566737342474/1128040123989242008

### CODE REVIEW

**Comment by leeduan** [comment]
@EasyEtHP4 / @kherzinger-figure FYI - relevant while you are still on the smart contract

File: smart-contract/src/contract.rs [hunk 1]
```diff
@@ -1484,10 +1648,31 @@ mod tests {
...
         let dcc = coin(1000, "dcc.coin");
         deps.querier
-            .base
-            .update_balance(Addr::unchecked("customer"), vec![dcc]);
-        deps.querier
-            .with_attributes("customer", &[("bank.kyc.pb", "ok", "string")]);
+            .mock_querier
+            .update_balance("customer", vec![dcc]);
+
+        // TODO - fix test since mock response returns same result no matter the input
```
Comment (leeduan) [inline smart-contract/src/contract.rs:1654]: waiting on resolution from provwasm team about mocking not considering the inputs of the query

File: smart-contract/src/contract.rs [hunk 2]
```diff
@@ -1739,9 +2010,30 @@ mod tests {
         )
         .unwrap();
 
+        // TODO - fix mock response so it is only returnable for checking bank2 and not bank1
```
Comment (leeduan) [inline smart-contract/src/contract.rs:2013]: waiting on resolution from provwasm team about mocking not considering the inputs of the query

File: smart-contract-sale/Cargo.toml [hunk 3]
```diff
@@ -28,16 +28,17 @@ overflow-checks = true
 backtraces = ["cosmwasm-std/backtraces"]
 
 [dependencies]
-cosmwasm-std = { version = "1.2.3", features = ["iterator"] }
-cosmwasm-storage = { version = "1.2.3", features = ["iterator"] }
+cosmwasm-std = { version = "1.2.6", features = ["iterator"] }
+cosmwasm-storage = { version = "1.2.6", features = ["iterator"] }
 cw2 = "1.0.1"
-provwasm-std = { version = "1.1.2" }
+provwasm-std = { version = "2.0.0-rc1" }
```
[the remark this answers is not in the document]
Comment (leeduan) [reply smart-contract-sale/Cargo.toml:34]: for some reason they never published it on crate...

File: smart-contract/src/contract.rs [hunk 4]
```diff
@@ -44,27 +50,42 @@ pub fn instantiate(
...
+                    amount: "0".to_string(),
+                }),
+                manager: env.contract.address.to_string(),
+                from_address: env.contract.address.to_string(),
+                status: MarkerStatus::Finalized.into(),
+                marker_type: MarkerType::Restricted.into(),
+                access_list: vec![
+                    AccessGrant {
+                        address: env.contract.address.to_string(),
+                        permissions: vec![1, 2, 3, 4, 5, 6, 7],
```
[the remark this answers is not in the document]
Comment (leeduan) [reply smart-contract/src/contract.rs:73]: kind of annoying even the migration guide isnt using the enum

**Review by jtalis — APPROVED** [review]

**Pull request merged**